cmd/fleetctl/generate_gitops.go

```diff
@@ -0,0 +1,1192 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	pathUtils "path"
+	"reflect"
+	"regexp"
+	"strings"
+	"unicode"
+
+	"github.com/fleetdm/fleet/v4/pkg/spec"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/fleetdm/fleet/v4/server/ptr"
+	"github.com/fleetdm/fleet/v4/server/service"
+	"github.com/ghodss/yaml"
+	"github.com/urfave/cli/v2"
+)
+
+type SecretWarning struct {
+	Filename string
+	Key      string
+}
+
+type Note struct {
+	Filename string
+	Note     string
+}
+
+type Messages struct {
+	SecretWarnings []SecretWarning
+	Notes          []Note
+}
+
+type Comment struct {
+	Filename string
+	Comment  string
+	Token    string
+}
+
+type FileToWrite struct {
+	Path    string
+	Content map[string]interface{}
+}
+
+type Software struct {
+	Hash       string
+	AppStoreId string
+	Comment    string
+}
+
+type teamToProcess struct {
+	ID   *uint
+	Team *fleet.Team
+}
+
+type generateGitopsClient interface {
+	GetAppConfig() (*fleet.EnrichedAppConfig, error)
+	GetEnrollSecretSpec() (*fleet.EnrollSecretSpec, error)
+	ListTeams(query string) ([]fleet.Team, error)
+	ListScripts(query string) ([]*fleet.Script, error)
+	ListConfigurationProfiles(teamID *uint) ([]*fleet.MDMConfigProfilePayload, error)
+	GetScriptContents(scriptID uint) ([]byte, error)
+	GetProfileContents(profileID string) ([]byte, error)
+	GetTeam(teamID uint) (*fleet.Team, error)
+	ListSoftwareTitles(query string) ([]fleet.SoftwareTitleListResult, error)
+	GetSoftwareTitleByID(ID uint) (*fleet.SoftwareTitle, error)
+	GetPolicies(teamID *uint) ([]*fleet.Policy, error)
+	GetQueries(teamID *uint, name *string) ([]fleet.Query, error)
+	GetLabels() ([]*fleet.LabelSpec, error)
+	Me() (*fleet.User, error)
+}
```

Comment:
This is an interface that the production `service.Client` satisfies, but can also be satisfied by a mock client in tests.